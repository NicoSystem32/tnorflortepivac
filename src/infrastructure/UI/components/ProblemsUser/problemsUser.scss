.content-link-service{
    text-align: center;
    margin-top: 50px;
    a{
        font-family: var(--font-main-regular);
        font-size: 16px;
        text-decoration: underline !important;
        color: var(--dominant-color-dark) !important;
    }
}
.content-options-pu{
    border: 1px solid var(--text-opacity4);
    border-radius: 10px;
    margin: 15px 5% 0;
    display: flex;
    padding: 18px 10px;
    gap: 10px;
    text-decoration: none !important;
    cursor: pointer;
}
.content-title {
    padding-top: 93px;
    padding-left: 5%;
    padding-right: 5%;
}
.customer-service {
    display: flex;
    padding: 30px 5%;
    gap: 10px;
    cursor: pointer;
    a{
        font-size: 16px;
        font-family: var(--font-main-regular);
        color: var(--dominant-color-dark) !important;
    }
}
.form-img{
    top: 130px;
    position: absolute;
    left: 6%;
}
.form-login-failed{
    width: 90%;
    margin: 82px auto 80px auto;
    min-height: 515px;
    box-shadow: 0px 3px 6px var(--text-opacity3);
    background-color: var(--background-color);
    z-index: 1;
}
.link-options-pu {
    font-size: 14px;
    font-family: var(--font-main-semibold);
    color: var(--text-opacity9) !important;
    margin: 0;
    display: flex;
    text-decoration: none;
}
.text-options-user{
    font-size: 14px;
    font-family: var(--font-main-semibold);
    color: var(--text-opacity9) !important;
    display: flex;
    text-decoration: none;
    white-space: nowrap;
}
.text-options-pu {
    font-family: var(--font-main-regular) !important;
    margin: 0;
}
.title-form {
    color: var(--text-opacity9);
    font-family: var(--font-main-semibold);
    font-size: 18px;
    margin: 0;
    display: flex;
}
.title-high {
    color: var(--dominant-color-dark);
    font-family: var(--font-main-semibold);
    font-size: 18px;
    display: flex;
    margin-bottom: 0 !important;
}

@media (max-width:321px) {
    .title-form {
        white-space: nowrap;
    }
}
//Desktop
@media (min-width:1180px) and (max-width: 1400px){
    .content-link-service{
        padding-bottom: 50px;
    }
}
//Giant
@media (min-width:1400px){
    .content-link-service{
        padding-bottom: 50px;
    }
}