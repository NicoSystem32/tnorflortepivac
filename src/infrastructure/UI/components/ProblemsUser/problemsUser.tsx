import Form from 'react-bootstrap/Form'

// Images
import {
    PointsPNG,
    PointsSVG,
    ElipsePNG,
    ElipseDeskPNG,
    WomenPNG,
    NetworksPNG,
    LoginPNG,
} from '../../utils/getImages'

// Icons
import { ForgotSVG } from '../../utils/getIcons'

// Styles
import './problemsUser.scss'

const ProblemsUser = (): JSX.Element => {
    return (
        <div className="content-global-login">
            <img src={PointsPNG} srcSet={PointsSVG} alt="" className="points-img" />
            <div className="content-global-init">
                <Form className="form-login-failed">
                    <img src={LoginPNG} alt="" className="form-img" />
                    <div className="content-title"></div>
                    <a
                        id="forget-password-pu"
                        className="content-options-pu"
                        href="/forget-password"
                    >
                        <img src={ForgotSVG} alt="forgot" className="img-options-pu" />
                        <p className="link-options-pu">
                            Olvidé<p className="text-options-pu">&nbsp;mi contraseña</p>
                        </p>
                    </a>
                    <a
                        id="locked-user-pu"
                        className="content-options-pu"
                        href="https://transacciones.cavipetrol.com/DwPortalApp/Cavionline/UI/Web/bloqueado_m.aspx?tabID=417&mID=3411"
                        target="_blank"
                        rel="noreferrer"
                    >
                        <img src={ForgotSVG} alt="forgot" className="img-options-pu" />
                        <p className="link-options-pu">
                            Desbloquear<p className="text-options-pu">&nbsp;mi usuario</p>
                        </p>
                    </a>
                    <a
                        id="new-user-pu"
                        className="content-options-pu"
                        href="https://transacciones.cavipetrol.com/DwPortalApp/Cavionline/UI/Web/x_crea0_m.aspx?tabID=417&mID=3411"
                        target="_blank"
                        rel="noreferrer"
                    >
                        <img src={ForgotSVG} alt="forgot" className="img-options-pu" />
                        <p className="link-options-pu">
                            Soy nuevo,<p className="text-options-pu">&nbsp;quiero registrarme</p>
                        </p>
                    </a>
                    <div className="content-link-service">
                        <a id="support-pu" href="/support">
                            Contactar a servicio al asociado
                        </a>
                    </div>
                </Form>
            </div>
            <img src={ElipsePNG} alt="" className="elipse-img" />
            <img src={ElipseDeskPNG} alt="" className="elipse-img-desk elipse-failed" />
            <img src={NetworksPNG} alt="" className="networks-img failed" />
            <img src={WomenPNG} alt="" className="women-img" />
        </div>
    )
}
export default ProblemsUser
