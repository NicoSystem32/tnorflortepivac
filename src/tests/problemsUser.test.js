import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import ProblemsUser from '../infrastructure/UI/components/ProblemsUser/problemsUser'

describe('Pruebas en <ProblemsUser />', () => {
    const wrapper = shallow(<ProblemsUser />)

    test('debe de mostrar <ProblemsUser /> correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('boton chatear debe redirigir correctamente', () => {
        expect(wrapper.find('#forget-password-pu').prop('href')).toBe('/forget-password')
    })

    test('boton chatear debe redirigir correctamente', () => {
        expect(wrapper.find('#locked-user-pu').prop('href')).toBe(
            'https://transacciones.cavipetrol.com/DwPortalApp/Cavionline/UI/Web/bloqueado_m.aspx?tabID=417&mID=3411'
        )
    })

    test('boton chatear debe redirigir correctamente', () => {
        expect(wrapper.find('#new-user-pu').prop('href')).toBe(
            'https://transacciones.cavipetrol.com/DwPortalApp/Cavionline/UI/Web/x_crea0_m.aspx?tabID=417&mID=3411'
        )
    })

    test('boton chatear debe redirigir correctamente', () => {
        expect(wrapper.find('#support-pu').prop('href')).toBe('/support')
    })
})
