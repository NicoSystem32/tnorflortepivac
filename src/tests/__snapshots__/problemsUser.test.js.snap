// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ProblemsUser /> debe de mostrar <ProblemsUser /> correctamente 1`] = `
<div
  className="content-global-login"
>
  <img
    className="points-img"
    src="points.png"
    srcSet="points.svg"
  />
  <div
    className="content-global-init"
  >
    <Form
      className="form-login-failed"
    >
      <img
        alt=""
        className="form-img"
        src="login.png"
      />
      <div
        className="content-title"
      />
      <a
        className="content-options-pu"
        href="/forget-password"
        id="forget-password-pu"
      >
        <img
          alt=""
          className="img-options-pu"
          src="forgot.png"
          srcSet="forgot.svg"
        />
        <p
          className="link-options-pu"
        >
          Olvidé
          <p
            className="text-options-pu"
          >
             mi usuario
          </p>
        </p>
      </a>
      <a
        className="content-options-pu"
        href="https://transacciones.cavipetrol.com/DwPortalApp/Cavionline/UI/Web/bloqueado_m.aspx?tabID=417&mID=3411"
        id="locked-user-pu"
        target="_blank"
      >
        <img
          alt=""
          className="img-options-pu"
          src="forgot.png"
          srcSet="forgot.svg"
        />
        <p
          className="link-options-pu"
        >
          Desbloquear
          <p
            className="text-options-pu"
          >
             mi usuario
          </p>
        </p>
      </a>
      <a
        className="content-options-pu"
        href="https://transacciones.cavipetrol.com/DwPortalApp/Cavionline/UI/Web/x_crea0_m.aspx?tabID=417&mID=3411"
        id="new-user-pu"
        target="_blank"
      >
        <img
          alt=""
          className="img-options-pu"
          src="forgot.png"
          srcSet="forgot.svg"
        />
        <p
          className="link-options-pu"
        >
          Soy nuevo,
          <p
            className="text-options-pu"
          >
             quiero registrarme
          </p>
        </p>
      </a>
      <div
        className="content-link-service"
      >
        <a
          href="/support"
          id="support-pu"
        >
          Contactar a servicio al asociado
        </a>
      </div>
    </Form>
  </div>
  <img
    alt=""
    className="elipse-img"
    src="elipse.png"
  />
  <img
    alt=""
    className="elipse-img-desk elipse-failed"
    src="elipse-desk.png"
  />
  <img
    alt=""
    className="networks-img failed"
    src="networks.png"
  />
  <img
    alt=""
    className="women-img"
    src="women.png"
  />
</div>
`;
